packages/ui-components/src/Charts/barChart.tsx

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import classnames from "classnames/bind";
+import {
+  BarChart as ReBarChart,
+  Bar,
+  XAxis,
+  YAxis,
+  Tooltip,
+  Legend,
+  ResponsiveContainer,
+  Cell,
+} from "recharts";
+import { Margin } from "recharts/types/util/types";
+
+import { ColorCoreNeutral0 } from "@cockroachlabs/design-tokens";
+import { METRIC_COLORS } from "./helpers";
+
+import styles from "./barChart.module.scss";
+const cx = classnames.bind(styles);
+
+export type Alignment = "vertical" | "horizontal";
+
+/**
+ * getColor is used to grab a color from a defined list of metric colors,
+ * so we can have a "semi-randomized" view of all our color schemes.
+ * @param index - Index of a metric color we want to use.
+ * @param customColors - Optional list of colors to use if available.
+ * @returns - Hex value representing a metric color.
+ */
+const getColor = (index: number, customColors?: string[]) => {
+  const colors = customColors || METRIC_COLORS;
+
+  // Default to the first color if we have an invalid index;
+  if (index < 0 || index >= colors.length) {
+    return colors[index % colors.length];
+  }
+  return colors[index];
+};
+
+interface BarChartProps<T> {
+  data: T[];
+  mainKey: keyof T;
+  dataKeys: (keyof T)[];
+  alignment: Alignment;
+  width: number | undefined;
+  height: number | undefined;
```

Comment:
Isn't the type of `number | undefined` the same an an optional parameter (`width?: number`)?

Reply:
ah yep! updated.